docs/the-git-based-process-you-need-for-mainframe-development-DRAFT/the-git-based-development-process-for-mainframe-development.md

```diff
@@ -0,0 +1,303 @@
+    - [Implement a fix of the current production state](#implement-a-fix-of-the-current-production-state)
+    - [Use epic branches for significant development initiatives](#use-epic-branches-for-significant-development-initiatives)
+  
+[Pipeline design and implementation supporting the workflows](pipeline-design-and-implementation-supporting-the-workflows.md#pipeline-design-and-implementation-supporting-the-workflows)
+
+[Conclusion](conclusion.md#conclusion)
+
+[Appendix](appendix.md#appendix)
+
+
+# The Git-based development process for Mainframe development
+
+As Git became the de-facto version control system in today's IT world, new terminologies such as *repositories*[^1], *branches*[^2], and *merges* arose. By agreeing upon a central Git server to integrate and consolidate changes, development teams were able to collaborate more efficiently and effectively. Building upon the open-source vanilla Git implementation, popular Git providers including GitHub, GitLab, and Bitbucket have implemented additional workflow features to facilitate a secure and stable development process. These include features such as *Pull Requests* (sometimes referred to as *Merge Requests*) to support coordination with Git in larger teams. The term *Pull Request* will be used throughout this document to designate the operation of reviewing and merging one branch into another.
+
+Many mainframe development teams follow a release-based or iteration-based process to deliver incremental updates to a pre-defined production runtime.
+
+## Characteristics of mainline-based development with feature branches
+
+As mentioned in the [Introduction](introduction.md) our recommended approach scales very well to support the needs of a range of
+team sizes, freqency and size of changes, and degrees of concurrent working.
+### Starting simple
+
+The mainline-based development approach[^3] with *feature* branches is a simple and structured workflow to implement, integrate, and deliver changes with an early integration process flow using a single long-living branch: *main*. Developers work in isolation in *feature* branches to implement changes to the source code, and ideally test the changes in a specific environment.
+
+This approach can be compared to a trunk-based branching model that leverages feature branches. A similar workflow like outlined in this publication is also documented by Microsoft without giving it a name[^5].
+
+<!-- this paragraph would make an ideal 'annotation', callout or sidebar if writing for an SSG like Mkdocs. -->
+(You may also see reference to *topic* branches, for example [here](https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows). 
+Feature branches can be considered as one sub-type of *topic* branches with branches for *bug-fixes* being another sub-type. 
+Both feature branches and bug-fix branches are expected to be relatively short-lived and ultimately to merge into a long-lived branch.)
```

Comment:
@ijmitch My understanding is that we didn't use "topic branches" throughout the document (instead of "feature branches") to avoid accidentally lumping in epic branches, but it seemed a little strange to me that we brought up the existence of the "bug-fix branch" sub-type, and then never mentioned it again. How about:

> (You may also see reference to *topic* branches, for example in Pro Git's [Git Branching - Branching Workflows](https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows) section. Feature branches can be considered as one sub-type of *topic* branches, with branches for bug-fixes being another sub-type.  Both feature branches and bug-fix branches are expected to be relatively short-lived and ultimately to merge into a long-lived branch. Although this document primarily references the term "feature branch", bug-fix branches can be managed similarly.)